mono/eglib/gmodule-win32.c

```diff
@@ -189,3 +192,155 @@ g_module_build_path (const gchar *directory, const gchar *module_name)
 	}
 	return g_strdup_printf ("%s%s" LIBSUFFIX, lib_prefix, module_name); 
 }
+
+// This is not about GModule but is still a close fit.
+// This is not named "g_" but that should be ok.
+// g_free the result
+// No MAX_PATH limit.
+//
+// Prefer mono_get_module_filename over mono_get_module_filename_ex and mono_get_module_basename.
+// Prefer not-ex, not-base.
+//
+gboolean
+mono_get_module_filename (gpointer mod, gunichar2** pstr, guint32* plength)
+{
+	gunichar2* str = NULL;
```

Comment:
```suggestion
	gunichar2 *str = NULL;
```

Reply:
It's kinda nonsense syntax widely reviled but whatever.